docs/curate-audio/process-data/audio-analysis/format-validation.md

```diff
@@ -0,0 +1,309 @@
+---
+description: "Validate audio file formats and detect corrupted files for robust audio curation pipelines"
+categories: ["processors"]
+tags: ["audio-validation", "format-support", "error-handling", "quality-control", "file-validation"]
+personas: ["data-scientist-focused", "mle-focused"]
+difficulty: "intermediate"
+content_type: "how-to"
+modality: "audio-only"
+---
+
+
+
+(audio-analysis-format-validation)=
+
+# Format Validation
+
+Verify audio file formats and detect corrupted files to ensure robust audio curation pipelines with reliable data quality.
+
+## Overview
+
+The format validation processor verifies audio file integrity, format compatibility, and accessibility before processing. This prevents pipeline failures and ensures consistent data quality across large audio datasets.
+
+:::note
+This page shows a custom approach to format validation. Classes such as `AudioFormatValidator` and `AudioValidationFilter` are examples and are not built-in stages. Create these as custom stages using `soundfile` for detection and validation, or adapt the example stage patterns from the overview page.
+:::
+
+## Key Features
+
+- **Format Detection**: Automatic audio format detection and validation
+- **Corruption Detection**: Identifies corrupted or unreadable audio files
+- **Compatibility Checking**: Ensures files are compatible with processing pipeline
+- **Error Reporting**: Detailed error logs for debugging and quality control
+
+## Supported Formats
+
+NeMo Curator supports formats compatible with the `soundfile` library (backed by `libsndfile`):
+
+- **WAV** - Uncompressed audio (recommended for high quality)
+- **FLAC** - Lossless compression with metadata support
+- **OGG** - Open-source compressed format
+- **MP3** - Compressed format (requires system dependencies)
+- **M4A** - Apple compressed format
+- **AIFF** - Apple uncompressed format
+
+## How it Works
+
+The format validation processor performs comprehensive checks on audio files:
+
+:::note
+Example: The following code uses `AudioFormatValidator` as a conceptual class. Create a similar custom stage (for example, a `LegacySpeechStage` that calls `sf.info(...)`) to populate validation fields such as `is_valid` and `validation_status` in your batch entries.
```

Comment:
Let's remove conceptual class examples. They are not very helpful with the current state of our codebase IMO.